sdk/core/core-http/lib/policies/userAgentPolicy.ts

```diff
@@ -15,6 +15,28 @@ import {
 
 export type TelemetryInfo = { key?: string; value?: string };
 
+/**
+ * Options for adding user agent details to outgoing requests.
+ */
+export interface UserAgentOptions {
+  /*
+   * String prefix to add to the user agent for outgoing requests.
+   * Defaults to an empty string.
+   */
+  userAgentPrefix?: string;
```

Comment:
The `userAgentPolicy()` that one would use to create the policy allows one to provide the value to be set in the user agent header where as here in `UserAgentOptions`, we are allowing to provide the prefix.

Meaning users trying to use the pipeline options can only set the prefix and users using the `userAgentPolicy()` **have** to provide the entire user agent string. 

There is an opportunity here to streamline the user agent story and exactly what we want to allow to be configured.

I want to suggest the below
- Do not allow customizing the name of the header to be used for User agent strings. Our guideline is specific as to what header is to be used. 
- Do not allow customizing what the value should be. Allow only the addition of the prefix.

If there is customer/library wanting to use their own header or their own value, they can do so by setting the request options or creating their own policy.

Code wise, this translates to
- `UserAgentOptions` only having the prefix support
- `userAgentPolicy()` updated to take `UserAgentOptions` as input

Reply:
My understanding is that the purpose of this setting is to allow the user to add their own tag to the front of the user agent header string, probably to identify their application.  That was just my assumption from the context, though; I have no idea if it's being used that way in practice.  I'm fine with whatever approach we want to take here.